pandas/core/groupby.py

```diff
@@ -4429,7 +4429,13 @@ def _groupby_indices(values):
         # bit better than factorizing again
         reverse = dict(enumerate(values.categories))
         codes = values.codes.astype('int64')
-        _, counts = _hash.value_count_int64(codes, False)
+        keys, counts = _hash.value_count_int64(codes, False)
```

Comment:
Per your suggestion, I borrowed some of the logic from Categorical.value_counts. The mask.all() check seemed to make the code run slower in this case, so I left it out.